src/test/java/org/apache/platypus/server/grpc/QueryTest.java

```diff
@@ -0,0 +1,414 @@
+                                .setMaxExpansions(1))
+                        .setAnalyzer(getTestAnalyzer())
+                        .setOperator(MatchOperator.MUST))
+                .build();
+
+        Consumer<SearchResponse> responseTester = searchResponse -> {
+            assertEquals(1, searchResponse.getTotalHits());
+            assertEquals(1, searchResponse.getHitsList().size());
+            SearchResponse.Hit hit = searchResponse.getHits(0);
+            String docId = hit.getFieldsMap().get("doc_id").getFieldValue(0).getTextValue();
+            assertEquals("2", docId);
+            checkHits(hit);
+        };
+
+        testQuery(querySupplier, responseTester);
+    }
+
+    @Test
+    public void testSearchMatchPhraseQuery() {
+        Supplier<Query> querySupplier = () -> Query.newBuilder()
+                .setQueryType(QueryType.MATCH_PHRASE)
+                .setMatchPhraseQuery(MatchPhraseQuery.newBuilder()
+                        .setField("vendor_name")
+                        .setQuery("SECOND second")
+                        .setSlop(1)).build();
+
+        Consumer<SearchResponse> responseTester = searchResponse -> {
+            assertEquals(1, searchResponse.getTotalHits());
+            assertEquals(1, searchResponse.getHitsList().size());
+            SearchResponse.Hit hit = searchResponse.getHits(0);
+            String docId = hit.getFieldsMap().get("doc_id").getFieldValue(0).getTextValue();
+            assertEquals("2", docId);
+            checkHits(hit);
+        };
+
+        testQuery(querySupplier, responseTester);
+    }
+
+    @Test
+    public void testSearchMatchPhraseQueryCustomAnalyzer() {
+        Supplier<Query> querySupplier = () -> Query.newBuilder()
+                .setQueryType(QueryType.MATCH_PHRASE)
+                .setMatchPhraseQuery(MatchPhraseQuery.newBuilder()
+                        .setField("vendor_name")
+                        .setQuery("<br> SECOND again </br>")
+                        .setAnalyzer(getTestAnalyzer())
+                        .setSlop(1)).build();
+
+        Consumer<SearchResponse> responseTester = searchResponse -> {
+            assertEquals(1, searchResponse.getTotalHits());
+            assertEquals(1, searchResponse.getHitsList().size());
+            SearchResponse.Hit hit = searchResponse.getHits(0);
+            String docId = hit.getFieldsMap().get("doc_id").getFieldValue(0).getTextValue();
+            assertEquals("2", docId);
+            checkHits(hit);
+        };
+
+        testQuery(querySupplier, responseTester);
+    }
+
+    /**
+     * Search with the query and then test the response. Additional test with boost will also be performed on the query.
+     *
+     * @param querySupplier {@link Supplier} that provides the query to test with
+     * @param responseTester {@link Consumer} that tests a {@link SearchResponse}
+     */
+    private void testQuery(Supplier<Query> querySupplier, Consumer<SearchResponse> responseTester) {
```

Comment:
Not sure I understand why we need a Supplier here since the supplier seems to return the static query. Seems like we could simply pass in `Query query` instead and the code would still work?

Reply:
The supplier will return a new query object every time its called. It will still work as expected if we pass just Query because the object is not being modified, the supplier was more of a precaution.

Reply:
I meant if we change this method signature

`private void testQuery(Supplier<Query> querySupplier, Consumer<SearchResponse> responseTester)`

to 

`private void testQuery(Query query, Consumer<SearchResponse> responseTester)`

This code should still work right?